pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -1346,7 +1346,38 @@ struct GatherOpConversion : public OpConversionPattern<tile::GatherOp> {
     // Return interpolation result (result = c0*g0 + c1*g1)
     auto p0 = rewriter.create<mlir::MulFOp>(loc, c0, g0).getResult();
     auto p1 = rewriter.create<mlir::MulFOp>(loc, c1, g1).getResult();
-    return rewriter.create<mlir::AddFOp>(loc, p0, p1).getResult();
+    auto result = rewriter.create<mlir::AddFOp>(loc, p0, p1).getResult();
+    switch (paddingMode) {
+    case PaddingMode::edge:
```

Comment:
This section of code needs refactoring, the edge padding mode code has to move here.